### DISCUSSION TITLE
[Bug][skywalking-client-js]broke when  I request a cross domain request （arraybuffer）

### DISCUSSION THREAD

**Discussion by Mikey-Lucky** [apache/skywalking#12760] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

NodeJS Client Side Agent (apache/skywalking-client-js)

### What happened

When I request a cross domain request,the console display that "Uncaught InvalidStateError: Failed to read the 'responseText' property from 'XMLHttpRequest': The value is only accessible if the object's 'responseType' is '' or 'text' (was 'arraybuffer')".The stack display the error is generated by ”Ajax Error Collection“.

### What you expected to happen

index.js:1 Uncaught InvalidStateError: Failed to read the 'responseText' property from 'XMLHttpRequest': The value is only accessible if the object's 'responseType' is '' or 'text' (was 'arraybuffer').
    at eval (index.js:1:6619)
    at XMLHttpRequest.f (index.js:1:22756)
    at XMLHttpRequest.eval (index.js:1:20573)

### How to reproduce

let xhr= new XMLHttpRequest();
xhr.open('GET','...',true);// blocked by CORS
xhr.responseType = 'arraybuffer';
xhr.send();

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Why a cross domain is reported to us?

**Reply by Mikey-Lucky** [reply]
The requested address comes from the server, and sometimes cross domain situations may occur in the testing environment. Actually, we don't know if one day it will appear in the production environment due to some mistakes.

**Reply by wu-sheng** [reply]
```
// Report collected data to `http:// + window.location.host + /browser/perfData` in default
ClientMonitor.register({
  # Use core/default/restPort in the OAP server.
  # If External Communication Channels are activated, `receiver-sharing-server/default/restPort`,
  # ref to https://skywalking.apache.org/docs/main/latest/en/setup/backend/backend-expose/
  collector: 'http://127.0.0.1:12800', 
  service: 'test-ui',
  pagePath: '/current/page/name',
  serviceVersion: 'v1.0.0',
});
```

This is the address the client agent is going to access. That is all we do. And you are the one adding this to suitable pages of your web app.

**Reply by Mikey-Lucky** [reply]
I'm very sorry, but I think you misunderstood. I did it this way, cross domain requests are not for collection purposes, but for our normal business requests. When our normal business requests cross domains and are 'arraybuffer', it will cause the collection requests to not work properly.

**Comment by Mikey-Lucky** [comment]
![1](https://github.com/user-attachments/assets/692e4224-3f63-41f5-ab70-0765cb64d7f7)

**Reply by wu-sheng** [reply]
This is actually not a cross-domain error. `http` request doesn't have that limitation, only https has that. 
The logs are very clear, `sw8` is a header skywalking client js agent injected to do distributed tracing, and the `dfhamds.cn` doesn't accept that. That is all. 

If this domain is not on your control, you should not add client-js to the page which posts this request.

**Reply by Mikey-Lucky** [reply]
We will handle this. but I don't think this exception should cause an error when collecting through Skywalking, right?

**Reply by Mikey-Lucky** [reply]
The screenshot I took is just one of the situations, in fact, knowing that problems will occur across domains will lead to them

**Reply by wu-sheng** [reply]
> We will handle this. but I don't think this exception should cause an error when collecting through Skywalking, right?

I don't know that part.

**Reply by Mikey-Lucky** [reply]
![2](https://github.com/user-attachments/assets/162e8d0f-d732-455c-ba00-5896adfe1a25)
The second mistake comes from Skywalking

**Reply by wu-sheng** [reply]
I saw that, but can't tell why is that.

**Reply by Mikey-Lucky** [reply]
I understand, thank you for your answer
